### ISSUE TITLE
Recent versions of Bitcoin Unlimited tank debug.log privacy features

### ISSUE THREAD

**Issue by 6784** [gandrewstone/bitcoin#472]
By default Bitcoin Core does not log IP addresses into the debug.log or console output to address privacy concerns when requesting log output from a user. Recent versions of "Bitcoin Unlimited" do not respect the option for some log prints, potentially compromising privacy where they would not have expected it. 

      -logips
           Include IP addresses in debug output (default: 0)

Log prints should be updated to respect this option, and not print IP addresses into logs by default.

**Comment by ftrader** [comment]
Good point if true. I don't suppose you have a concrete example of such a trace from the code?

I'm running up a node with logips=0 now to check, and we'll obviously go over the code to check for any instances.

Messages found so far that would seem to fit the bill (IPs redacted with [IP]):

`Node [IP] socket recv error 'Connection reset by peer (104)`

`connect() to [IP]:[port] failed: Network is unreachable (101)`

These are clearly network-specific error messages, so it might make sense to display IP information if displaying these at all. Will need to check whether they were introduced by BU, but they are certainly quite verbose and present in logs.

I guess there is also this one which would have to be dropped entirely if IP logging is not enabled:

`ProcessMessages: advertising address [IP:port]`

Apart from these, I haven't seen any other ones in my log from a node running for a while.
Code inspection to follow.

**Comment by ftrader** [comment]
After a day's running of the node, found:

`socket send error 'No route to host (113)' to [IP:port] (3043)`

`socket send error 'Connection timed out (110)' to [IP:port] (2918)`

`Misbehaving: [IP:port] (0 -> 100) BAN THRESHOLD EXCEEDED`
